vspace/src/main/java/edu/asu/diging/vspace/core/aspects/ExhibitionDataAspect.java

```diff
@@ -48,7 +63,104 @@ public void setExhibition(JoinPoint jp) {
                         ((Model) obj).addAttribute("publishedSpaces", publishedSpaces);
                     }
                 }
+            }  
+        }
+    }
+
+    @Around("execution(public * edu.asu.diging.vspace.web.publicview..*Controller.*(..))")
+    public Object showExhibition(ProceedingJoinPoint jp) throws Throwable {
+        Object[] args = jp.getArgs();
+        MethodSignature signature = (MethodSignature) jp.getSignature();
+        int indexOfModel = (Arrays.asList(signature.getParameterTypes())).indexOf(Model.class);
+        Exhibition exhibition = (Exhibition) exhibitionManager.getStartExhibition(); 
+        // If there is no exhibition, we go back to root url page.
+        if(exhibition==null) {
+            return "redirect:/";
+        }
+        //If no exhibition mode has been setup for existing exhibition, we skip modes and aspects.
+        if(exhibition.getMode() == null) {
+            return jp.proceed();
+        }
+        Map<IdPrefix, String> ids = getIds(args, signature);
+        String spaceId = ids.getOrDefault(IdPrefix.SPACEID, "");
+        String moduleId = ids.getOrDefault(IdPrefix.MODULEID, "");
+        return redirectRequest(jp, spaceId, moduleId, indexOfModel, exhibition);
+    }
+    
+    
+    
+    // Based on exhibition mode, get the redirect page or pass control to controller.  
+    public Object redirectRequest(ProceedingJoinPoint jp, String spaceId, String moduleId, int modelIndex, Exhibition exhibition) throws Throwable{
+        ISpace space = spaceManager.getSpace(spaceId);
+        IModule module = moduleManager.getModule(moduleId);
+        Object[] args = jp.getArgs();
+        // If exhibition is set to offline, set the custom message or default message.
+        if(exhibition.getMode().equals(ExhibitionModes.OFFLINE)) {
+            String modeValue = exhibition.getCustomMessage().equals("") == false ? exhibition.getCustomMessage() : exhibition.getMode().getValue();
+            ((Model) args[modelIndex]).addAttribute("modeValue", modeValue);
+        }
+        // If exhibition is set to maintenance, set the default message.
+        if(exhibition.getMode().equals(ExhibitionModes.MAINTENANCE)) {
+            ((Model) args[modelIndex]).addAttribute("modeValue", exhibition.getMode().getValue());
+        }
+        // If user is not logged in and exhibition is not active, show maintenance page.
+        if(authFacade.getAuthenticatedUser()==null && !exhibition.getMode().equals(ExhibitionModes.ACTIVE)) {
```

Comment:
you've used `exhibition.getMode()` so often now, that I would put it in a variable